[2024-09-26T21:37:45.152+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-26T21:37:45.169+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_job_dag.submit_spark_job manual__2024-09-26T21:32:39.953054+00:00 [queued]>
[2024-09-26T21:37:45.178+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_job_dag.submit_spark_job manual__2024-09-26T21:32:39.953054+00:00 [queued]>
[2024-09-26T21:37:45.179+0000] {taskinstance.py:2865} INFO - Starting attempt 2 of 2
[2024-09-26T21:37:45.194+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): submit_spark_job> on 2024-09-26 21:32:39.953054+00:00
[2024-09-26T21:37:45.198+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=273) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-09-26T21:37:45.200+0000] {standard_task_runner.py:72} INFO - Started process 274 to run task
[2024-09-26T21:37:45.200+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'spark_job_dag', 'submit_spark_job', 'manual__2024-09-26T21:32:39.953054+00:00', '--job-id', '157', '--raw', '--subdir', 'DAGS_FOLDER/jobsSpark.py', '--cfg-path', '/tmp/tmpatfyr3p3']
[2024-09-26T21:37:45.202+0000] {standard_task_runner.py:105} INFO - Job 157: Subtask submit_spark_job
[2024-09-26T21:37:45.244+0000] {task_command.py:467} INFO - Running <TaskInstance: spark_job_dag.submit_spark_job manual__2024-09-26T21:32:39.953054+00:00 [running]> on host 38a711029e2e
[2024-09-26T21:37:45.311+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_job_dag' AIRFLOW_CTX_TASK_ID='submit_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2024-09-26T21:32:39.953054+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-09-26T21:32:39.953054+00:00'
[2024-09-26T21:37:45.312+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-26T21:37:45.333+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-09-26T21:37:45.335+0000] {spark_submit.py:403} INFO - Spark-Submit cmd: spark-submit --master spark://spark:7077 --num-executors 2 --executor-memory 1G --driver-memory 512M --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py
[2024-09-26T21:37:46.935+0000] {spark_submit.py:579} INFO - python3: can't open file '/opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py': [Errno 2] No such file or directory
[2024-09-26T21:37:46.945+0000] {spark_submit.py:579} INFO - 24/09/26 21:37:46 INFO ShutdownHookManager: Shutdown hook called
[2024-09-26T21:37:46.947+0000] {spark_submit.py:579} INFO - 24/09/26 21:37:46 INFO ShutdownHookManager: Deleting directory /tmp/spark-bf4207ff-b325-4478-8b24-ead1e1929aff
[2024-09-26T21:37:46.982+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --num-executors 2 --executor-memory 1G --driver-memory 512M --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py. Error code is: 2.
[2024-09-26T21:37:46.986+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=spark_job_dag, task_id=submit_spark_job, run_id=manual__2024-09-26T21:32:39.953054+00:00, execution_date=20240926T213239, start_date=20240926T213745, end_date=20240926T213746
[2024-09-26T21:37:47.066+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-26T21:37:47.069+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 157 for task submit_spark_job (Cannot execute: spark-submit --master spark://spark:7077 --num-executors 2 --executor-memory 1G --driver-memory 512M --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py. Error code is: 2.; 274)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark:7077 --num-executors 2 --executor-memory 1G --driver-memory 512M --name arrow-spark --deploy-mode client /opt/bitnami/spark/src/scripts_spark/WebScraping_Transformation.py. Error code is: 2.
[2024-09-26T21:37:47.101+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-09-26T21:37:47.122+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-26T21:37:47.126+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
